csharp/ql/src/Stubs/Stubs.qll

```diff
@@ -513,12 +513,30 @@ private string stubClassName(Type t) {
                                   stubClassName(element), "," order by i
                                 ) + ")"
                         else
-                          if t instanceof ValueOrRefType
+                          if t instanceof FunctionPointerType
                           then
-                            result =
-                              stubQualifiedNamePrefix(t) + t.getUndecoratedName() +
-                                stubGenericArguments(t)
-                          else result = "<error>"
+                            exists(CallingConvention callconvention, string calltext |
```

Comment:
Perhaps add `FunctionPointerType fpt`, assign `fpt = this`, and replace the three `t.(FunctionPointerType)` with `fpt`.

Reply:
Yes, I thought about that as well, but decided against it because last time I did it, the review comment recommend to do it the other way around 😄  But on the other hand, here we already have an exists, so it will make it nicer 👍

Reply:
It makes sense to introduce a new variable (either with `exists` or `any`) when there are multiple identical casts, as in this case.

Reply:
Great - that is the rule of 👍 I needed 😸